pinot-core/src/main/java/org/apache/pinot/core/operator/filter/ExpressionFilterOperator.java

```diff
@@ -196,7 +196,7 @@ public BlockDocIdIterator iterator() {
       int _endDocId;
       //used only in next() and advance methods
       int _currentBlockStartDocId = -1;
-      int _currentBlockEndDocId = 0;
+      int _currentBlockEndDocId = -1;
```

Comment:
I would recommend making both endDocId and this exclusive. Currently the numDocsScanned value is incorrect.